### PR TITLE
Fix bug related to folding and hash comments

### PR DESCRIPTION
**Pull request by alexconst**
Fix bug that occurs when hash comments are used in fenced code blocks.
Fix same problem that occurs when building a TOC with the :Toc command.
Add flag that allow the user to set folding level, by default set to 1.

### CODE REVIEW

**Comment by shirosaki** [comment]
Looks great.
Can you make test cases for that like the following?
https://github.com/plasticboy/vim-markdown/blob/master/test/folding.vader

**Comment by shirosaki** [comment]
`:Toc` doesn't make setext headers. Folding works fine with setext headers.
http://spec.commonmark.org/0.23/#setext-headings

**Comment by alexconst** [comment]
I guess I found out how deep the rabbit hole goes on this one. More than I wished for.

Long story short: it works, although technically it's not correct, but this already comes from the master branch.
~~~markdown
# chap 1 (this has a foldlevel of 0)
hello
# chap 2 (while this has a foldlevel of 1 but it should be 0)
bye
~~~

I initially tried to make things correct and pretty (ie, more useful at least to me) which you would end up with something like this:
~~~text
  1 +-- 14 lines: # chap 1---------
 15 +-- 15 lines: ## chap 1.1------
 30 +--  4 lines: # chap 2---------
 34 +--  7 lines: ## chap 2.1------
 41 +--  4 lines: # chap 3---------
 45 +--  5 lines: chap 4-----------     setext header
 50 +-- 10 lines: chap 4.1---------    setext header
~~~
The problem was that despite everything working fine, the tests failed (even cascading to code that I haven't touched). On occasion even when manually checking and the value was equal to the one in .vader, the test failed.

So long story short: last commit gets the job done and passing in tests, but is not analytically correct.

**Comment by shirosaki** [comment]
Thank you! Merged.

Test failures on Travis is due to the terminal window width is small on Travis.
I fixed some test failures.

I'm not sure about the bug that foldlevel should be 0.

Single `=` and `-` are valid for setex header so that I fixed this part of `:Toc`.
http://spec.commonmark.org/0.23/#example-47

**Comment by alexconst** [comment]
@shirosaki 
Thanks for merging them.

You probably know this but just in case, folding is broken in `master` because of the `s:is_mkdCode` function.
What exactly is the purpose of that function?

**Comment by shirosaki** [comment]
`s:is_mkdCode` is introduced by fixing `#` folding in code block.
It is same issue as you fixed. See #73.
It tries to detect code block region from syntax.
If you don't have `syntax on` setting, it doesn't work.

**Comment by alexconst** [comment]
My problem was that I had `syntax=ghmarkdown`. I have disabled that config and everything works.
Thanks.
